app-server/app/src/main/scala/bucket/Bucket.scala

```diff
@@ -0,0 +1,152 @@
+package com.azavea.rf.bucket
+
+import java.sql.Timestamp
+import java.util.{Date, UUID}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+import com.azavea.rf.datamodel.latest.schema.tables._
+import com.azavea.rf.utils.Database
+
+
+trait Bucket {
+
+  implicit val database:Database
+  implicit val ec:ExecutionContext
+
+  import database.driver.api._
+
+  /** Add bucket to database
+    *
+    * @param bucket BucketsRow bucket to add to database
+    */
+  def insertBucket(bucket: BucketsRow): Future[Try[BucketsRow]] = {
+    database.db.run {
+      Buckets.forceInsert(bucket).asTry
+    } map {
+      case Success(_) => Success(bucket)
+      case Failure(e) => Failure(e)
+    }
+  }
+
+  /** Get scenes that belong to a bucket
+    *
+    * @param bucketId UUID bucket to request scenes for
+    */
+  def getBucketScenes(bucketId: UUID): Future[Seq[ScenesRow]] = {
+    val bucketSceneQuery = for {
+      bucketToScene <- ScenesToBuckets if bucketToScene.bucketId === bucketId
```

Comment:
Is there a performance difference between this and `<- ScenesToBuckets.filter(_.bucketId === bucketId)`?